pilot/pkg/model/config.go

```diff
@@ -1039,11 +1047,26 @@ func (store *istioConfigStore) AuthenticationPolicyByDestination(hostname Hostna
 			out = spec
 		}
 	}
-	// Zero-currentMatchLevel implies no config matching the destination found.
-	if currentMatchLevel == 0 {
-		return nil
+	// Non-zero currentMatchLevel implies authentication policy was found for the given host.
+	if currentMatchLevel != 0 {
+		return &out
 	}
-	return &out
+
+	// Reach here if no authentication policy found in service or namespace level; check for global policy.
+	// Note that, global authentication policy must be:
+	// - In global config namespace (i.e `istio-global-config`)
+	// - Name is `global`. Policies with different name are ignored.
+	//
+	// Also, the global authentication policy should not have any targets (as it's supposed to
```

Comment:
Do we want to check this in the code below? I.e., no target in the spec.

Reply:
I think ignoring is safer for now.

All these would be nicer if we can check and reject policy at CRD time. Unfortunately, the current istio config framework doesn't include config metadata in the validation callback. We can do some improvement on this later.